pkg/config/deployment.go

```diff
@@ -220,9 +224,13 @@ type WaitStageOptions struct {
 
 // WaitStageOptions contains all configurable values for a WAIT_APPROVAL stage.
 type WaitApprovalStageOptions struct {
+	// The maximum length of time to wait before giving up.
+	Timeout   Duration `json:"timeout"`
 	Approvers []string `json:"approvers"`
 }
 
+var defaultWaitApprovalTimeout = Duration(6 * time.Hour)
```

Comment:
nit, should move to the top of this file.

Reply:
For this I'm still debating. I'm trying to figure out where to define the default values. 

Currently most of all default values are populated at the pakcages where they are used. We have not debated this point until now, but I followed that rule because this current design can hide the way how to use each configuration field and leave it to each package that uses them.

But on the dark side, it can't guarantee that all field is populated when it parsed. Happy to hear your thoughts.

Reply:
> Currently most of all default values are populated at the packages where they are used.

Got it, I think I'm okay with this convention 👍 What I mentioned here is, should we move this default value to the top of this file and make it const, currently it's been used before declared make it hard to read.
For the point of "where to define the default value", I think we should debate and address it by another PR. 😁

Reply:
> should we move this default value to the top of this file and make it const,
> I think we should debate and address it by another PR. 

I am fine with these things. To be honest I was thinking about where to define and how to configure the default value of our config. And I think all of them should be defined in the `config` package (as public consts) and loaded when the configuration is parsed. So the comment about the default value is the same in that package too. The docs refer to them too.

It would be nice if we can define the default value by go tag (as JSON tag). There are some libs for doing that thing but still not found a good enough one.
Let me make a PR for this issue.

Reply:
> It would be nice if we can define the default value by go tag (as JSON tag).

👍 👍 👍

Reply:
Okay, let's debate about all of them on another PR. 

>should we move this default value to the top of this file and make it const, 

@sanposhiho Could you address it if no objection to that? 😄

Reply:
I've just caught up on this conversation. OK, move the default value to the top 👍